cosmic_ray/mutating.py

```diff
@@ -49,26 +49,30 @@ def activation_record(self):
         """
         return self._activation_record
 
-    def visit_mutation_site(self, node, op):
+    def visit_mutation_site(self, node, op, count):
         """Potentially mutate `node`, returning the mutated version.
 
         `Operator` calls this when AST iteration reaches a
         potential mutation site. If that site is scheduled for
         mutation, the subclass instance will be asked to perform the
         mutation.
         """
-        if self._count == self._target:
-            self._activation_record = {
-                'operator': _full_module_name(op),
-                'occurrence': self._target,
-                'line_number': cosmic_ray.util.get_line_number(node)
-            }
-
-            old_node = node
-            node = op.mutate(old_node)
-            ast.copy_location(node, old_node)
-
-        self._count += 1
+        for idx in range(count):
```

Comment:
Yeah, it is getting a bit complex! When operators were only allowed to make one mutation per node, `self._target` represented the N-th time an operator encountered a node that it could mutate. Now `self._target` represents the N-th time an operator could perform a mutation, taking into account that an operator might make multiple mutations per node.

Similarly, `self._count` now represents how many mutations an operator has so far claimed it could make while walking the AST.

So the basic idea of the change is to say this: if the operator claims that it performs enough mutations on the current node to reach the target, then let it do it the mutation. 

Maybe the thing to do is to get your version working and stable. Then we can clean up the code later.